.commitlintrc.yml

```diff
@@ -2,32 +2,79 @@
-    - refactor
-    - perf
-    - test
-    - revert
   body-max-line-length:
     - 2
     - always
     - 200
+  type-enum:
+    - 2
+    - always
+    - - chore
+      - ci
+      - docs
+      - feat
+      - fix
+      - perf
+      - refactor
+      - revert
+      - style
+      - test
+      - theme
 help: |
   **Possible types**:
-  `chore`:    Change build process, tooling or dependencies.
-  `ci`:       Changes to our CI configuration files and scripts
-  `feat`:     Adds a new feature.
-  `fix`:      Solves a bug.
-  `docs`:     Adds or alters documentation.
+
+
+  `chore`:    Changes that involve maintenance or updates to build tools and dependencies.
+  
+              **Usage**: Use for routine tasks or updates that don’t modify source or test files.
+              **Example**: `chore: update dependency versions`
+
+  `ci`:       Updates to CI/CD configuration files and scripts.
+  
+              **Usage**: Use for modifications that affect continuous integration and delivery.
+              **Example**: `ci: add deployment script`
+
+  `docs`:     Changes to documentation files.
+  
+              **Usage**: Use for updates or corrections to documentation.
+              **Example**: `docs: update API documentation`
+
+  `feat`:     New features or enhancements.
+  
+              **Usage**: Use for introducing new functionalities or enhancements.
+              **Example**: `feat: add user authentication`
+
+  `fix`:      Bug fixes or corrections.
+  
+              **Usage**: Use for resolving bugs or issues.
+              **Example**: `fix: correct typo in README`
+
+  `perf`:     Performance improvements.
+  
+              **Usage**: Use for changes that enhance performance.
+              **Example**: `perf: optimize image loading`
+
+  `refactor`: Code improvements that don’t affect functionality.
+  
+              **Usage**: Use for restructuring or improving code without changing its behavior.
+              **Example**: `refactor: simplify component logic`
+
+  `revert`:   Reverts previous changes or commits.
+  
+              **Usage**: Use for undoing changes from earlier commits.
+              **Example**: `revert: undo changes from commit abc123`
+
+  `style`:    Code style and formatting changes.
+  
+              **Usage**: Use for changes that affect code style or formatting but not functionality.
+              **Example**: `style: format code with prettier`
+
+  `test`:     Additions or corrections to tests.
+  
+              **Usage**: Use for new tests or fixes to existing tests.
+              **Example**: `test: add unit tests for user service`
+
   `theme`:    Work on or add a theme.
-  `refactor`: Rewrites code without feature, performance or bug changes.
-  `perf`:     Improves performance.
-  `test`:     Adds or modifies tests.
-  `revert`:   Changes that reverting other changes
+  
+              **Usage**: Use for theme-related changes.
+              **Example**: `theme: add new theme`
```

Comment:
adding a new theme isn't allowed, you can only edit existing ones. SO best to reflect that in the help as well.

Reply:
> adding a new theme isn't allowed, you can only edit existing ones. SO best to reflect that in the help as well.

I’m not quite sure what you mean. Are you suggesting I should completely change the help section, including the examples? Or are you referring to modifying the `theme` type and its help description? Or is it about removing the new `style` type I added? I believe the `style` type is somewhat necessary, as it focuses on code style changes without altering functionality, which differs slightly from `refactor`.

Reply:
The example states `theme: add new theme` but that isn't allowed. So it should be something like `theme(jandedobbeleer): update git branch icon`.